examples/ml/example_custom_algorithm.py

```diff
@@ -595,14 +595,22 @@ def training_step(self, batch: BatchedTrainingSamples) -> BatchedTrainingOutputs
             metrics=metrics,
         )
 
-    def configure_optimizers(self) -> list[torch.optim.Optimizer]:
+    def configure_optimizers(
```

Comment:
Considering that optimizers do not need num_training_steps, I suggest you split this function, and we have a `configure_optimizers`, and a `configure_schedulers(optimizers: list[torch.optim.Optimizer]`